src/pages/items-page/hooks/useResponsivePageSize.js

```diff
@@ -0,0 +1,35 @@
+import { useState, useEffect, useCallback } from "react";
+import { breakpoints } from "@constants/breakpoints";
+
+/**
+ * 화면 너비에 따라 서버에서 받아올 데이터의 개수(pageSize)를 설정하는 커스텀 훅
+ * @param {{ desktop: number, tablet: number, mobile: number }} config - 각 해상도별 pageSize 값
+ * @param {Function} [onChange] - pageSize setter
+ */
+export const useResponsivePageSize = (config, onChange) => {
+  const [pageSize, setPageSize] = useState(1);
+
+  const updatePageSize = useCallback(() => {
+    const width = window.innerWidth;
+    let newSize;
+
+    if (width >= parseInt(breakpoints.desktop)) {
+      newSize = config.desktop;
+    } else if (width >= parseInt(breakpoints.tablet)) {
+      newSize = config.tablet;
+    } else {
+      newSize = config.mobile;
+    }
+
+    setPageSize(newSize);
+    if (onChange) onChange(newSize);
```

Comment:
💬 여담
옵셔널 체이닝을 통해 아래처럼도 작성 가능합니다!

```suggestion
    onChange?.(newSize);
```
https://developer.mozilla.org/ko/docs/Web/JavaScript/Reference/Operators/Optional_chaining